impl/events.go

```diff
@@ -169,10 +195,26 @@ func (m *manager) OnResponseReceived(chid datatransfer.ChannelID, response datat
 func (m *manager) OnRequestTimedOut(ctx context.Context, chid datatransfer.ChannelID) error {
 	log.Warnf("channel %+v has timed out", chid)
 
+	m.reconnectsLk.Lock()
+	reconnect, ok := m.reconnects[chid]
+	var alreadyReconnected bool
+	select {
+	case <-reconnect:
+		alreadyReconnected = true
+	default:
+	}
+	if !ok || alreadyReconnected {
+		reconnect = make(chan struct{})
+		m.reconnects[chid] = reconnect
+	}
+	m.reconnectsLk.Unlock()
+	timer := time.NewTimer(m.channelRemoveTimeout)
+
 	go func() {
 		select {
 		case <-ctx.Done():
```

Comment:
Do we need to do the channel cleanup in this case?

Reply:
We have to cancel the context when a channel restarts, so this is important to do.